src/AddToCalendarApiWidget.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace Drupal\addtocalendar;
+
+use Drupal\Core\Datetime\DrupalDateTime;
+
+/**
+ * Class AddToCalendarApiWidget.
+ */
+class AddToCalendarApiWidget {
+  /**
+   * Declare various setting variables for add to calendar api widget.
+   *
+   * @var string
+   */
+  protected $atcStyle;
+  protected $atcDisplayText;
+  protected $atcTitle;
+  protected $atcDescription;
+  protected $atcLocation;
+  protected $atcOrganizer;
+  protected $atcOrganizerEmail;
+  protected $atcDateStart;
+  protected $atcDateEnd;
+  protected $atcPrivacy;
+  protected $atcDataSecure;
+  protected $timeZone;
+  /**
+   * Hold the various calendars usable in the widget.
+   *
+   * @var array
+   */
+  protected $atcDataCalendars = [];
+
+  /**
+   * Constructs a new AddToCalendarApiWidget object.
+   */
+  public function __construct() {
+
+    $this->atcStyle = 'blue';
+    $this->atcDisplayText = 'Add to calendar';
+    $this->atcTitle = 'Some event title';
+    $this->atcDescription = 'Some event description';
+    $this->atcLocation = 'Some event location';
+    // Fetching site name and site email id.
+    $config = \Drupal::config('system.site');
+    $site_name = $config->get('name');
+    $site_mail = $config->get('mail');
+
+    $this->atcOrganizer = $site_name;
+    $this->atcOrganizerEmail = $site_mail;
+    $this->atcDateStart = 'now';
+    $this->atcDateEnd = 'now';
+    $this->atcPrivacy = 'public';
+    $this->atcDataSecure = 'auto';
+    $data_calendars = array('iCalendar',
+      'Google Calendar',
+      'Outlook',
+      'Outlook Online',
+      'Yahoo! Calendar',
+    );
+    $this->atcDataCalendars = $data_calendars;
+    $this->timeZone = date_default_timezone_get();
+
+  }
+
+  /**
+   * Use this function to set values for the widget.
+   */
+  public function setWidgetValues($config_values = array()) {
+    foreach ($config_values as $key => $value) {
+      $this->$key = $value;
+    }
+  }
+
+  /**
+   * Use this function to get a particular value from this widget.
+   */
+  public function getWidgetValue($config_value) {
+    return $this->$config_value;
+  }
+
+  /**
+   * Constructs and returns a renderable array widget for add to calendar.
+   */
+  public function generateWidget() {
+
+    // Start building the renderable array.
+    $build['addtocalendar'] = [];
+    $display_text = t('%text', array('%text' => $this->atcDisplayText));
+    $build['addtocalendar_button'] = [
+      '#type' => 'html_tag',
+      '#tag' => 'a',
+      '#value' => $display_text->__toString(),
+      '#attributes' => [
+        'class' => 'atcb-link',
+      ],
+    ];
+
+    $timeZone = $this->timeZone;
+    $date = new DrupalDateTime(preg_replace('/T/', ' ', $this->atcDateStart), $timeZone);
```

Comment:
If you are hoping to get atcDateStart from user or database, this might not be good. As user/database value's timezone is UTC. and additional setTimezone() should be used after creating the DrupalDateTime object.

But this should be fine when atcDateStart = 'now'. So should handle both the cases.